### PR TITLE
3.1.x fixes

### PR DESCRIPTION
**Pull request by basdelfos**
Fixes incompatibility with BF 3.1.x.

- Fixed Motor tab.
- Hide Power & Battery tab when pre-BF 3.2.0.
- Added old battery / current configuration back for pre-BF 3.2.0 in Configuration tab.
- Bumped version number to 3.1.3 and added version information.
- Hide reverse motor direction checkbox for pre-BF 3.2.0 (and show correct motor preview on Motor tab).

I've tested this with 2 quads running BF 3.1.7 and BF 3.2.0 (master from this afternoon) and both seem to work now. If someone want to test it also before we release this version, that would be great.

### CODE REVIEW

File: _locales/en/messages.json [hunk 1]
```diff
@@ -2480,6 +2480,45 @@
     "configurationMagHardware": {
         "message": "Magnetometer (if supported)"
     },
+    // ---------------------------------------------
```
Comment (mikeller) [inline _locales/en/messages.json]: This is not valid JSON. Please remove. Also, there are a lot of other messages in here that will have to be removed when the functionality to support older versions gets removed, so marking these specifically does not buy us anything.

File: js/msp/MSPHelper.js [hunk 2]
```diff
@@ -192,46 +193,63 @@ MspHelper.prototype.process_data = function(dataHandler) {
...
             case MSPCodes.MSP_VOLTAGE_METER_CONFIG:
-                VOLTAGE_METER_CONFIGS = [];
-                var voltage_meter_count = data.readU8();
-
-                for (var i = 0; i < voltage_meter_count; i++) {
-                    var subframe_length = data.readU8();
-                    if (subframe_length != 5) {
-                        for (var j = 0; j < subframe_length; j++) {
-                            data.readU8();
+                if (semver.lt(CONFIG.apiVersion, "1.36.0")) {
```
Comment (mikeller) [inline js/msp/MSPHelper.js]: We probably need to check here for the firmware being Betaflight and not cleanflight, not to break compatibility. @hydra would know more.

File: manifest.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
     "manifest_version": 2,
     "minimum_chrome_version": "38",
-    "version": "3.1.2",
+    "version": "3.1.3",
```
Comment (mikeller) [inline manifest.json]: Should be 3.2.0.

File: tabs/configuration.html [hunk 4]
```diff
@@ -489,6 +489,111 @@
             <div class="clear-both"></div>
         </div>
         
+
+        <!-- -------------------------------------------------------- -->
+        <!-- Remove this part when we officialy retire BF 3.1.x       -->
```
Comment (mikeller) [inline tabs/configuration.html]: Again, no point in marking this part specifically.

File: tabs/configuration.js [hunk 5]
```diff
@@ -140,13 +148,39 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
...
+        }
+
         if (semver.gte(CONFIG.apiVersion, "1.20.0")) {
             MSP.send_message(MSPCodes.MSP_NAME, false, false, next_callback);
         } else {
             next_callback();
         }
     }
 
+    // -----------------------------------------------
```
Comment (mikeller) [inline tabs/configuration.js]: This is confusing. If only what you have marked as 'remove' is removed, line 153 will cause an exception.

File: tabs/configuration.js [hunk 6]
```diff
@@ -140,13 +148,39 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
...
+    // Remove this when we officily retire BF 3.1.x
+    function load_battery() {
+        var next_callback = load_current;
+        if (semver.gte(CONFIG.flightControllerVersion, "3.1.0")) {
```
Comment (mikeller) [inline tabs/configuration.js]: Should be `ApiVersion`.

File: tabs/motors.js [hunk 7]
```diff
@@ -42,7 +42,14 @@ TABS.motors.initialize = function (callback) {
         $('#content').load("./tabs/motors.html", process_html);
     }
 
-    MSP.send_message(MSPCodes.MSP_MOTOR_CONFIG, false, false, get_arm_status);
+    // Get information from Betaflight
+    if (semver.gte(CONFIG.apiVersion, "1.36.0")) {
+        // BF 3.2.0+
```
Comment (mikeller) [inline tabs/motors.js]: Again, probably needs to be gated to not break cleanflight.

File: tabs/motors.js [hunk 8]
```diff
@@ -167,8 +174,14 @@ TABS.motors.initialize = function (callback) {
         lines.attr('d', graphHelpers.line);
     }
 
-    function update_model(val) {
-        $('.mixerPreview img').attr('src', './resources/motor_order/' + mixerList[val - 1].image + '.svg');
+    function update_model(mixer) {
```
Comment (mikeller) [inline tabs/motors.js]: This seems to be unrelated to the battery configuration. Should this be in here, or should it be a separate PR?

**Comment by Brisingrox** [comment]
I will test this directly on 2 different quads and flight controller. FortiniF4 and KakuteF4

File: _locales/en/messages.json (hunk 1 above)
Comment (basdelfos) [inline _locales/en/messages.json]: Removed, as all the other messages.

File: js/msp/MSPHelper.js (hunk 2 above)
Comment (basdelfos) [inline js/msp/MSPHelper.js]: Didn't think of that. But maybe Cleanflight users also want to use the GUI to change battery and current settings on older fw versions. @hydra can you look at this, we want to push a new version of the BFC to the chome store as soon as possible to fix all the incompatibility issues with BF 3.1.x

File: manifest.json (hunk 3 above)
Comment (basdelfos) [inline manifest.json]: Done

File: tabs/configuration.js (hunk 6 above)
Comment (basdelfos) [inline tabs/configuration.js]: Copy-paste this over from the code that has been removed when then new Power & Battery tab was created.
